### PR TITLE
feat(arm): ACRGeoreplicated

### PR DESCRIPTION
**Pull request by SaraWeinberg1234**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

adds a ACRGeoreplicated

### Description
Checks whether a certain field exists and contains a specific value.
 If so, checks whether there is also a certain field that contains true

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/arm/checks/resource/ACRGeoreplicated.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.arm.base_resource_check import BaseResourceCheck
+
+
+class ACRGeoreplicated(BaseResourceCheck):
+    def __init__(self) -> None:
+
+        name = "Ensure geo-replicated container registries to match multi-region container deployments."
+        id = "CKV_AZURE_165"
+        supported_resources = ("Microsoft.ContainerRegistry/registries",)
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/ACRGeoreplicated.py:12]: Looks like u should scan the `Microsoft.ContainerRegistry/registries/replications` resource and look for the `location` field.

**Review by ChanochShayner — CHANGES_REQUESTED** [review]

File: checkov/arm/checks/resource/ACRGeoreplicated.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    def __init__(self) -> None:
+
+        name = "Ensure geo-replicated container registries to match multi-region container deployments."
+        id = "CKV_AZURE_165"
+        supported_resources = ("Microsoft.ContainerRegistry/registries/replications",)
+        categories = (CheckCategories.NETWORKING,)
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf: dict[str, list[Any]]) -> CheckResult:
+        sku = conf.get("sku")
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/ACRGeoreplicated.py:17]: i don't see SKU value here - 
https://learn.microsoft.com/en-us/azure/templates/microsoft.containerregistry/registries/replications?pivots=deployment-language-arm-template

Comment (SaraWeinberg1234) [reply checkov/arm/checks/resource/ACRGeoreplicated.py]: Please check it

**Review by ChanochShayner — APPROVED** [review]

File: checkov/arm/checks/resource/ACRGeoreplicated.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+from checkov.common.models.enums import CheckCategories
+from checkov.arm.base_resource_value_check import BaseResourceValueCheck
+
+
+class ACRGeoreplicated(BaseResourceValueCheck):
+    def __init__(self) -> None:
+
+        name = "Ensure geo-replicated container registries to match multi-region container deployments."
+        id = "CKV_AZURE_165"
+        supported_resources = ("Microsoft.ContainerRegistry/registries/replications",)
```
Comment (tsmithv11) [inline checkov/arm/checks/resource/ACRGeoreplicated.py:13]: @SaraWeinberg1234 @ChanochShayner I believe this needs to be a graph check that ensures `Microsoft.ContainerRegistry/registries` resources are connected to `Microsoft.ContainerRegistry/registries/replications`. location is a [required](https://learn.microsoft.com/en-us/azure/templates/microsoft.containerregistry/registries/replications?pivots=deployment-language-arm-template) field, so I believe this check will always pass (the fail UT is not valid).

**Review by tsmithv11 — CHANGES_REQUESTED** [review]

File: checkov/arm/checks/resource/ACRGeoreplicated.py (hunk 3 above)
Comment (ChanochShayner) [reply checkov/arm/checks/resource/ACRGeoreplicated.py]: If this should be a graph check let's close this PR, I don't think we currently support edges in ARM.

**Comment by ChanochShayner** [comment]
Closing by this - 
https://github.com/bridgecrewio/checkov/pull/6324#discussion_r1663545183

**Pull request closed without merging**